File: random/libs/random/src/random.cpp
#include <random.h>

std::vector<double> Random::GenerateSequence(int a, int m, int r0, int length) {
    std::vector<double> sequence(length);
    long long rn = r0;
    std::generate(sequence.begin(), sequence.end(), [a, m, &rn]() {
        rn = rn * a % m;
        return (double)rn / (double)m;
    });
    return sequence;
}

File: distribution/distribution/random.go
package distribution

import "math/rand"

type Random struct {
	length int
}

func (r Random) Generate() []float64 {
	dist := make([]float64, r.length)
	for i := range dist {
		dist[i] = rand.Float64()
	}
	return dist
}

func CreateRandomDistribution(length int) Distribution {
	return &Random{length: length}
}
